### ISSUE TITLE
Missing Supervisor.Spec worker - is it needed?

### ISSUE THREAD

**Issue by darrenklein** [circles-learning-labs/ecto_adapters_dynamodb#49]
In the Ecto 2 version of the adapter, under the main module's `child_spec/2` definition, we imported `Supervisor.Spec` and set up a `worker`:

```
    import Supervisor.Spec
    child_spec = worker(__MODULE__, [repo, opts])

    ecto_dynamo_log(:debug, "#{inspect __MODULE__}.child_spec", %{"#{inspect __MODULE__}.child_spec-params" => %{repo: repo, child_spec: child_spec, opts: opts}})
```

Under Ecto 3, `child_spec/2` is no longer called, and some of the other behaviors under that function were moved into `init/1` - however, the section above was not.

Its absence does not seem to adversely affect the adapter - should this be reinstated under Ecto 3, or is it not important?

@alhambra1 would be interested in your thoughts on this.

**Comment by alhambra1** [comment]
What do you mean by "however, the section above was not." ? It wasn't
moved? Who did or did not move what from where to where?

**Comment by darrenklein** [comment]
Under Ecto 3, `child_spec/2` was no longer being called, so I had commented it out. At some point, I deleted it, having forgotten that I hadn't revisited all of its functionality - the block above was overlooked, so it is not represented anywhere in the current Ecto 3 logic - so other logic was moved into a new place in the current implementation (`init/1`), but this bit wasn't.

_I_ did not move it from _`child_spec`_ to anywhere else. 

_I_ do not know what role this played in our initial implementation, so I wanted to know if you knew its purpose.

**Comment by alhambra1** [comment]
I don't know. Without it, does the adapter under Ecto 2 break?

>

**Comment by darrenklein** [comment]
Yes, the adapter under Ecto 2 needs it - after some further investigation, I think we're good to leave this out.  First off, it turns out that `Supervisor.Spec` was deprecated as of Elixir version 1.6... second, our main module's `init/1` is itself called from Ecto's `Ecto.Repo.Supervisor` module, so it looks like whatever supervision needs we had to implement ourselves under Ecto 2 are now handled for us under Ecto 3.

Closing this issue.
